app/src/main/java/com/portfolio/tasky/entry/fragments/LoginFragment.kt

```diff
@@ -33,24 +35,85 @@ class LoginFragment : Fragment(), FragmentInflater by FragmentInflaterImpl() {
             openRegistrationFragment()
         }
     }
+    private fun setFieldValidations(loginFragment: LoginFragment?) {
+        clearValidatorParams()
+
+        val etEmail = viewBinding.etEmail.subLayout.etInput
+        etEmail.addTextChangedListener(loginFragment?.let {
+            TaskyValidationWatcherImpl(viewBinding.etEmail, LOGIN,
+                it
+            )
```

Comment:
Here you'd just call the viewmodel.onEmailChange or something

And the viewmodel would just call the emailvalidator

Reply:
Correct. Implemented as per this.